bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/GoveeHandler.java

```diff
@@ -0,0 +1,336 @@
+                }
+            }
+            if (!thing.getStatus().equals(ThingStatus.ONLINE)) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error.could-not-query-device [\"" + goveeConfiguration.hostname
+                            + "\"]");
+        }
+    }
+
+    /**
+     * Initiate a refresh to our thing devicee
+     *
+     */
+    private void triggerDeviceStatusRefresh() throws IOException {
+        if (refreshRunning) {
+            return;
+        }
+        if (GoveeDiscoveryService.isDiscoveryActive()) {
+            logger.debug("Not triggering refresh as Scan is currently active");
+            return;
+        }
+        refreshRunning = true;
+
+        logger.debug("trigger Refresh Status of device {}", thing.getLabel());
+
+        try {
+            GenericGoveeRequest lightQuery = new GenericGoveeRequest(
+                    new GenericGoveeMsg("devStatus", new EmptyValueQueryStatusData()));
+            send(GSON.toJson(lightQuery));
+        } finally {
+            refreshRunning = false;
+        }
+    }
+
+    public void send(String message) throws IOException {
+        DatagramSocket socket;
+        socket = new DatagramSocket();
+        socket.setReuseAddress(true);
+        byte[] data = message.getBytes();
+
+        InetAddress address = InetAddress.getByName(goveeConfiguration.hostname);
+        logger.trace("Sending {} to {}", message, goveeConfiguration.hostname);
+        DatagramPacket packet = new DatagramPacket(data, data.length, address, SENDTODEVICE_PORT);
+        socket.send(packet);
+        socket.close();
+    }
+
+    public void updateDeviceState(@Nullable StatusResponse message) {
+        if (message == null) {
+            return;
+        }
+
+        logger.info("Update Device State ----------------------------------------------");
+        int lastOnOff = message.msg().data().onOff();
+        logger.info("lastOnOff = {}", lastOnOff);
+        lastBrightness = message.msg().data().brightness();
+        logger.info("lastbrigthess = {}", lastBrightness);
+        lastColor = message.msg().data().color();
+        logger.info("lastColor = {}", lastColor);
+
+        int lastColorTemperature = message.msg().data().colorTemInKelvin();
+
+        logger.info("Last RGB = {} {} {}", lastColor.r(), lastColor.g(), lastColor.b());
+        HSBType hsbColor = ColorUtil.rgbToHsb(new int[] { lastColor.r(), lastColor.g(), lastColor.b() });
+        logger.info("Last HSB = {} {} {}", hsbColor.getHue(), hsbColor.getSaturation(), hsbColor.getBrightness());
+
+        if (lastOnOff == 1) {
+            logger.info("setting BRIGHTNESS on Color to be consistent by using lastBrightness = {}", lastBrightness);
+            hsbColor = ColorUtil.rgbToHsb(new int[] { lastColor.r(), lastColor.g(), lastBrightness });
```

Comment:
Maybe this could be a cause of some of the problems you have mentioned. Brightness is provided as green (which is missing).

Reply:
Thanks, that is a very good catch. Indeed, I read **h s** b and gave **r g** B.
New Hope ;-)